app/controllers/v3/service_instances_controller.rb

```diff
@@ -424,6 +437,16 @@ def unprocessable_service_plan!
     unprocessable!('Invalid service plan. Ensure that the service plan exists, is available, and you have access to it.')
   end
 
+  def unavailable_service_plan!(service_plan, service_plan_guid)
+    unprocessable!("Invalid service plan. The service plan '#{service_plan}' with guid '#{service_plan_guid}' has been removed from the service broker's catalog." \
+                   'It is not possible to create new service instances using this plan.')
+  end
+
+  def service_plan_not_visible_in_space!(service_plan, service_plan_guid, space_name, space_guid)
+    unprocessable!("Invalid service plan. Ensure that the service plan '#{service_plan}' with guid '#{service_plan_guid}' \
```

Comment:
Should we make this message more clear? I.e. mention that this could be due to a space-scoped broker offering this plan or that the plan is not enabled in this org?